cms/db/filecacher.py

```diff
@@ -543,6 +545,33 @@ def _create_directory_or_die(directory):
             logger.error(msg)
             raise RuntimeError(msg)
 
+    def precache_lock(self):
+        """Lock the (shared) cache for precaching if it is currently unlocked.
+
+        The purpose is to avoid wasting resources by ensuring that on each
+        machine, only one worker precaches at any time.
+
+        return (fileobj|None): The lock file if the cache was previously
+            unlocked. Closing the file object will release the lock.
+            None if the cache was already locked.
+
+        """
+        lock_file = os.path.join(self.file_dir, "cache_lock")
+        fobj = open(lock_file, 'wb')
```

Comment:
Minor note: we don't need binary mode here.

Reply:
Okay, I changed it to 'w'.